lib/labels_web/controllers/page_controller.ex

```diff
@@ -33,14 +35,28 @@ defmodule LabelsWeb.PageController do
              owner: target_owner,
              repo_name: target_repo
            }) do
+      # create ok changeset -> keep source info, remove target info
       conn
       |> put_flash(:info, "Labels synched!")
-      |> redirect(to: Routes.page_path(conn, :index))
+      |> render(
+        "index.html",
+        [repositories: repositories] ++
+          form_values(source_owner: source_owner, source_repo: source_repo)
+      )
     else
       {{:error, :not_found}, repo} ->
         conn
         |> put_flash(:error, "#{repo} repository not found")
-        |> redirect(to: Routes.page_path(conn, :index))
+        |> render(
+          "index.html",
+          [repositories: repositories] ++
+            form_values(
+              source_owner: source_owner,
+              source_repo: source_repo,
+              target_owner: target_owner,
+              target_repo: target_repo
+            )
```

Comment:
When there is an error (ie target or source repo not found) we keep all the values entered by the user in the form